script/tasks/diamondUpdateFacet.sh

```
@@ -85,8 +85,17 @@ diamondUpdateFacet() {
-  # determine full (relative) path of deploy script
-  SCRIPT_PATH=$DEPLOY_SCRIPT_DIRECTORY"$SCRIPT.s.sol"
+  # Handle ZkSync specific paths and extensions
+  if isZkEvmNetwork "$NETWORK"; then
+    ZK_SCRIPT_PATH="script/deploy/zksync/$SCRIPT.zksync.s.sol"
+    # Extract contract name once, will be used multiple times later
+    CONTRACT_NAME=$(basename "$ZK_SCRIPT_PATH" | sed 's/\.zksync\.s\.sol$//')
+    # Check if the foundry-zksync binaries exist, if not fetch them
+    install_foundry_zksync
+  else
+    SCRIPT_PATH=$DEPLOY_SCRIPT_DIRECTORY"$SCRIPT.s.sol"
+    CONTRACT_NAME=""  # Not needed for non-zkEVM networks
```

Shellcheck (0.10.0)

[warning] CONTRACT_NAME appears unused. Verify use (or export if used externally).

(SC2034)